pulp_python/app/tasks/publish.py

```diff
@@ -0,0 +1,167 @@
+    """
+    publisher = python_models.PythonPublisher.objects.get(pk=publisher_pk)
+    repository = models.Repository.objects.get(pk=repository_pk)
+    latest_version = models.RepositoryVersion.latest(repository)
+
+    log.info(_('Publishing: repository={repo}, version={version}, publisher={pub}').format(
+        repo=repository.name,
+        version=latest_version.number,
+        pub=publisher.name
+    ))
+
+    with WorkingDirectory():
+        with models.Publication.create(latest_version, publisher) as publication:
+            write_simple_api(publication)
+
+    log.info(_('Publication: {pk} created').format(pk=publication.pk))
+
+
+def write_simple_api(publication):
+    """
+    Writes metadata mimicking the simple api of PyPI for all python packages in the
+    repository version
+
+    https://wiki.python.org/moin/PyPISimple
+
+    Args:
+        publication (pulpcore.plugin.models.Publication): A publication to generate metadata for
+    """
+
+    os.mkdir('simple')
+    project_names = python_models.PythonPackageContent.objects.order_by('name')\
+        .values_list('name', flat=True).distinct()
+
+    # generate a canonical name for each project name - the canonical name is all-lowercase
+    # with the names replaced by underscores
+    index_names = [(name, sanitize_name(name)) for name in project_names]
+
+    # write the root index, which lists all of the projects for which a package available
+    index_path = 'simple/index.html'
+    with open(index_path, 'w') as index:
+        context = Context({
+            'projects': index_names
+        })
+        template = Template(simple_index_template)
+        index.write(template.render(context))
+
+    index_md = models.PublishedMetadata(
+        relative_path=index_path,
+        publication=publication,
+        file=File(open(index_path, 'rb'))
+    )
+    index_md.save()
+
+    for (name, canonical_name) in index_names:
+        project_dir = 'simple/{}'.format(canonical_name)
+        os.mkdir(project_dir)
+
+        packages = python_models.PythonPackageContent.objects.filter(name=name)
+
+        package_detail_data = []
+
+        for package in packages:
+            filename = package.filename
+            artifact_set = package.contentartifact_set.all()
+            for content_artifact in artifact_set:
+                published_artifact = models.PublishedArtifact(
+                    relative_path=content_artifact.relative_path,
+                    publication=publication,
+                    content_artifact=content_artifact)
+                published_artifact.save()
+
+                md5sum = content_artifact.artifact.md5
+                path = "../../{}".format(filename)
+                package_detail_data.append((filename, path, md5sum))
+
+        metadata_relative_path = '{project_dir}/index.html'.format(project_dir=project_dir)
+
+        with open(metadata_relative_path, 'w') as simple_md:
```

Comment:
s/simple_md/simple_metadata/

Lets avoid abrevs if where it doesn't make a difference :)